### PR TITLE
Open Images + DIV2K dataset + testing updates

### PR DESCRIPTION
**Pull request by mmuckley**
This PR introduces the DIV2K and Open Images datasets. It also introduces a few testing updates for `utils.py`, in particular splitting apart the logic for creating a random image vs. writing it to a temporary file.

### CODE REVIEW

File: neuralcompression/data/_openimages_v6.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        super().__init__()
+        if split not in ("train", "val", "test"):
+            raise ValueError("Split must be in ('train', 'val', 'test')")
+
+        self.root = Path(root)
+        self.split = self.split_map[split]
+        self.transform = transform
+
+        if image_list_file is None:
+            self.image_list_file = self.root / f"list-{self.split}-files.txt"
```
Comment (QuentinDuval) [inline neuralcompression/data/_openimages_v6.py:65]: Should you instead provide the option to provide the paths to the splits ? This is what we do for Omnivision - it is more verbose but helps building alternate splits more easily for experiments

File: tests/data/test_div2k.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import pytest
+import torch
+from torchvision.transforms import ToTensor
+from utils import create_random_image, write_image_to_file
+
+from neuralcompression.data import DIV2KDataset
+
+
+@pytest.fixture
+def div2k_datafolder(tmp_path: Path):
```
Comment (QuentinDuval) [inline tests/data/test_div2k.py:18]: Oh nice !

File: tests/data/test_open_images.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    def write_folder_and_flist(base_dir: Path, subfolder: str):
+        directory = base_dir / subfolder
+        directory.mkdir(parents=True)
+
+        n = int(rng.integers(1, 16, (1,)))
+
+        images = []
+
+        with open(base_dir / f"list-{subfolder}-files.txt", "w") as f:
```
Comment (QuentinDuval) [inline tests/data/test_open_images.py:31]: To be adapted if you adapt the code to provide arbitrary split paths

File: neuralcompression/data/_openimages_v6.py (hunk 1 above)
Comment (mmuckley) [reply neuralcompression/data/_openimages_v6.py:65]: There was actually a bug here - the `image_list_file` was not used at all!

The user can provide a path to the splits I think, provided it's a subfolder with the name `'train'`, `'val'`, or `'test'`, but I think what we can do here is open a PR in the future for just a generic indexed image dataset that doesn't include this split prescription.

File: tests/data/test_open_images.py (hunk 3 above)
Comment (mmuckley) [reply tests/data/test_open_images.py:32]: Made a note for the future!

**Pull request merged**